File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day18Test.java
package com.capital7software.aoc2015.days;

import com.capital7software.aoc.aoc2015.days.Day18;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day18Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day18Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day18();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void animateLights() {
        var instance = new Day18();
        int expected = 4;

        var actual = instance.animateLights(4, lines, false);
        assertEquals(expected, actual.first(), "The number of lights on " +
                "is not what was expected: " + expected);
    }

    @Test
    void animateLightsCornersAlwaysOn() {
        var instance = new Day18();
        int expected = 17;

        var actual = instance.animateLights(5, lines, true);
        assertEquals(expected, actual.first(), "The number of lights on " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day18Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day18Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day18Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day18()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testRecoverFrequency() {
    val instance = Day18()
    val expected = 4L
    val actual = instance.recoverFrequency(lines)

    Assertions.assertEquals(
        expected, actual, "The number of the recovered frequency " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testGetMessageSentCount() {
    setupFromFile("inputs/input_day_18-02.txt")
    val instance = Day18()
    val expected = 3
    val actual = instance.getMessageSentCount(lines)

    Assertions.assertEquals(
        expected, actual, "The number of messages sent " +
        "is not what was expected: $expected"
    )
  }

}
